### PR TITLE
First pass at a guide for compiling macros to augmentations.

### PR DESCRIPTION
**Pull request by munificent**
This is a fairly high level imperative process for how a Dart
implementation could process macros and generate an augmentation. I
didn't want to go into too much unnecessary detail, but hopefully it's
concrete enough.

cc @leafpetersen @bwilkerson @johnniwinther

### CODE REVIEW

File: working/macros/feature-specification.md [hunk 1]
```diff
@@ -687,35 +687,134 @@ have the following restrictions:
...
+
+Instances of the `Identifier` class have special serialization. An import for
+library that the identifier resolves to is added to the augmentation with a new
+unique prefix identifier. The `Identifier` name is then serialized as a prefixed
+identifier for that prefix. (A more sophisticated implementation could reuse
+imports when multiple identifiers resolve to the same library, and may choose to
+omit the prefix entirely if the resulting identifier will still resolve
+correctly.)
+
+This augmentation library is written on disk in some implementation-defined
```
Comment (jakemac53) [inline working/macros/feature-specification.md:801]: I think that probably only the analyzer will actually write this to disk? Or possibly the CFE would in certain modes, but not others?

```suggestion
This augmentation library may be written on disk in some implementation-defined
```

**Review by jakemac53 — APPROVED** [review]

File: working/macros/feature-specification.md [hunk 2]
```diff
@@ -687,35 +687,134 @@ have the following restrictions:
...
 
-**TODO**: Update to use library cycles.
+Here's a (non-normative) illustration of how a Dart implementation could compile
+a Dart program containing macro applications:
 
-1.  For each library, ordered topologically by imports:
+#### 1. Break the program into library cycles
 
-    1.  For each declaration, with nested declarations ordered first:
+Starting at the entrypoint library, traverse all imports and exports to collect
```
[the remark this answers is not in the document]
Comment (munificent) [reply working/macros/feature-specification.md:703]: I intended those to be covered by "imports", but, yes, called those out explicitly.

File: working/macros/feature-specification.md [hunk 3]
```diff
@@ -687,35 +687,134 @@ have the following restrictions:
...
 
-    1.  Now all macros have been applied, all imperative code exists, and the
-        library can be completely compiled. Any macros defined in this library
-        are ready to be used by later libraries.
+#### 3. Apply macros
+
+In a sandbox environment or isolate, create an instance of the corresponding
+macro class for each macro application. Pass in any macro application arguments
+to the macro's constructor. If a parameter's type is `Code` or a subclass,
+convert the argument expression to a `Code` object.
```
[the remark this answers is not in the document]
Comment (munificent) [reply working/macros/feature-specification.md:739]: Good call. Done.

File: working/macros/feature-specification.md (hunk 1 above)
Comment (munificent) [reply working/macros/feature-specification.md:801]: Yeah, good catch. Applied locally.

**Pull request merged**